### PR TITLE
feat: support lazy loaded routes in `OktaAuthGuard`

### PR DESCRIPTION
**Pull request by shuowu**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](/okta/okta-angular/blob/master/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: OKTA-408692

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/okta/okta.guard.ts [hunk 1]
```diff
@@ -45,6 +51,20 @@ export class OktaAuthGuard implements CanActivate, CanActivateChild {
...
 
+  async canLoad(route: Route, segments: UrlSegment[]): Promise<boolean> {
+    this.onAuthRequired = route.data?.onAuthRequired || this.onAuthRequired;
+
+    const isAuthenticated = await this.oktaAuth.isAuthenticated();
+    if (isAuthenticated) {
+      return true;
+    }
+
+    const originalUri = segments[0].path;
```
Comment (oleksandrpravosudko-okta) [inline src/okta/okta.guard.ts:62]: is `baseHref`(whenever it is set) part of `segments[0]` too?

File: README.md [hunk 2]
```diff
@@ -164,6 +164,24 @@ const appRoutes: Routes = [
 ]
 ```
 
+You can use `canLoad` to achieve lazy loading for not immediately necessary modules to keep the initial bundle size smaller.
```
Comment (oleksandrpravosudko-okta) [inline README.md:167]: nit: not immediately loading?

Comment (shuowu) [reply README.md:167]: How about `for modules that are not immediately necessary`?

File: src/okta/okta.guard.ts (hunk 1 above)
Comment (shuowu-okta) [reply src/okta/okta.guard.ts:62]: It's relative path, no baseHref is included.

**Review by oleksandrpravosudko-okta — APPROVED** [review]

**Pull request closed without merging**